common/datasets/switchboard.py

```diff
@@ -0,0 +1,198 @@
+from i6_core.datasets.switchboard import *
+from i6_core.lexicon.conversion import LexiconFromTextFileJob
+from i6_core.lexicon.modification import WriteLexiconJob, MergeLexiconJob
+from i6_core.lib import lexicon
+
+
+def get_speakers_list(subdir_prefix=""):
+    """
+    Returns speakers list
+
+    :param str subdir_prefix: alias name prefix
+    :rtype: tk.Path
+    """
+    alias_name = os.path.join(subdir_prefix, "Switchboard")
+
+    speakers_stats = DownloadSwitchboardSpeakersStatsJob()
+    speakers_stats.add_alias(os.path.join(alias_name, "download_speakres_stats_job"))
+
+    speakers_list = CreateSwitchboardSpeakersListJob(speakers_stats.out_file)
+    speakers_list.add_alias(os.path.join(alias_name, "create_speakers_list_job"))
+
+    return speakers_list.out_speakers_list
+
+
+def get_train_bliss_corpus(audio_dir, subdir_prefix=""):
+    """
+    Returns Switchboard training bliss corpus
+
+    :param tk.Path audio_dir: path for audio data
+    :param str subdir_prefix: alias name prefix
+    :rtype: tk.Path
+    """
+    alias_name = os.path.join(subdir_prefix, "Switchboard")
+
+    swb_trans_and_dict = DownloadSwitchboardTranscriptionAndDictJob()
+    swb_trans_and_dict.add_alias(
+        os.path.join(alias_name, "download_trans_and_dict_job")
+    )
+
+    speakers_list = get_speakers_list(subdir_prefix=subdir_prefix)
+    corpus = CreateSwitchboardBlissCorpusJob(
+        audio_dir=audio_dir,
+        trans_dir=swb_trans_and_dict.out_trans_dir,
+        speakers_list_file=speakers_list,
+    )
+    corpus.add_alias(os.path.join(alias_name, "create_train_corpus_job"))
+
+    return corpus.out_corpus
+
+
+def get_special_lemma_lexicon():
+    """
+    Generate the special phonemes/lemmas for Switchboard
+
+    :rtype lexicon.Lexicon
+    """
+    lex = lexicon.Lexicon()
+
+    tags = ["[SILENCE]", "[NOISE]", "[VOCALIZED-NOISE]", "[LAUGHTER]"]
+    tag_to_phon = {
+        "[SILENCE]": "[SILENCE]",
+        "[NOISE]": "[NOISE]",
+        "[VOCALIZED-NOISE]": "[VOCALIZEDNOISE]",
+        "[LAUGHTER]": "[LAUGHTER]",
+    }
+    for tag in tags:
+        lex.add_phoneme(tag_to_phon[tag], variation="none")
+
+    # add non-special lemmas
+    for tag in tags[1:]:  # silence is considered below
+        lex.add_lemma(
+            lexicon.Lemma(
+                orth=[tag],
+                phon=[tag_to_phon[tag]],
+                synt=[[]],
+                eval=[[]],
+            )
+        )
+
+    # create special lemmas
+    lex.add_lemma(
+        lexicon.Lemma(
+            orth=["[SENTENCE-END]"], synt=[["</s>"]], special="sentence-boundary"
```

Comment:
any special reason this has no `eval=[[]]`?

Reply:
hmm not sure why. I checked again the current lexicon we use and it is without `eval`.

Reply:
the `sentence-end` lemma has the same `synt` token and there we have empty `eval` tag.